File: ivan/ql-project/src/net/iplantevin/ql/ast/types/ITypeVisitor.java
package net.iplantevin.ql.ast.types;

/**
 * @author Ivan
 */
public interface ITypeVisitor<T> {
    public T visit(BooleanType booleanType);

    public T visit(IntegerType integerType);

    public T visit(StringType stringType);

    public T visit(UndefinedType undefinedType);
}

File: rperz/QSLib/QSLib/Visitors/ITypeVisitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using QSLib.AST.Types;

namespace QSLib.Visitors
{
    public interface ITypeVisitor<T>
    {
        T Visit(IntegerType type);
        T Visit(StringType type);
        T Visit(BoolType type);
    }
}
